Crash upon database disconnect

I restarted postgres and chaind crashed due to the connection bouncing. It looks like this happened in the blocks service.

I'm happy to take on the bugfix PR for this, should be a good first contribution. Could you please confirm that you'd be open to such a PR?

```
$ ./chaind --version
0.4.2

...
{"level":"error","service":"blocks","impl":"standard","slot":2183551,"block_root":"0x38c0eca808e4348e60eeb217d1c40875e064c722fa2ac170c297227626b1b4c4","error":"failed to fetch metadata: failed to begin t
ransaction: FATAL: terminating connection due to administrator command (SQLSTATE 57P01)","time":"2021-09-30T20:30:36+02:00","message":"Failed to obtain metadata"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xcc1b7b]

goroutine 856 [running]:
github.com/wealdtech/chaind/services/blocks/standard.(*Service).OnBeaconChainHeadUpdated(0xc0001d0000, 0x120be10, 0xc0000b2700, 0x21517f, 0x8e34e408a8ecc038, 0x7508c4d117b2ee60, 0x70c12afa22c764e0, 0xc4b
4b126762297c2, 0xe6b5f372d2826080, 0x48ccb0ceae7a91c7, ...)
        /home/henridf/chaind/services/blocks/standard/handler.go:59 +0x69b
github.com/wealdtech/chaind/services/blocks/standard.(*Service).updateAfterRestart.func1(0xc006e88000)
        /home/henridf/chaind/services/blocks/standard/service.go:173 +0xe5
github.com/attestantio/go-eth2-client/http.(*Service).handleEvent(0xc0001f9810, 0xc00bf5c080, 0xc00001e540)
        /home/henridf/go/pkg/mod/github.com/attestantio/go-eth2-client@v0.7.1/http/events.go:155 +0x24d
github.com/attestantio/go-eth2-client/http.(*Service).Events.func1.1(0xc00bf5c080)
        /home/henridf/go/pkg/mod/github.com/attestantio/go-eth2-client@v0.7.1/http/events.go:68 +0x3d
github.com/r3labs/sse/v2.(*Client).SubscribeWithContext.func1(0x0, 0x0)
        /home/henridf/go/pkg/mod/github.com/r3labs/sse/v2@v2.3.0/client.go:89 +0x17e
gopkg.in/cenkalti/backoff%2ev1.RetryNotify(0xc0088a7ec8, 0x11fdfd0, 0xc00059e2a0, 0x0, 0xc0005e4550, 0xc000798754)
        /home/henridf/go/pkg/mod/gopkg.in/cenkalti/backoff.v1@v1.1.0/retry.go:37 +0x1bc
github.com/r3labs/sse/v2.(*Client).SubscribeWithContext(0xc0001eb050, 0x120be10, 0xc0000b2700, 0x0, 0x0, 0xc000798770, 0x1, 0xde6801)
        /home/henridf/go/pkg/mod/github.com/r3labs/sse/v2@v2.3.0/client.go:99 +0x125
github.com/r3labs/sse/v2.(*Client).SubscribeRawWithContext(...)
        /home/henridf/go/pkg/mod/github.com/r3labs/sse/v2@v2.3.0/client.go:228
github.com/attestantio/go-eth2-client/http.(*Service).Events.func1(0xc0001eb050, 0x120be10, 0xc0000b2700, 0xc0001f9810, 0xc00001e540)
        /home/henridf/go/pkg/mod/github.com/attestantio/go-eth2-client@v0.7.1/http/events.go:67 +0x170
created by github.com/attestantio/go-eth2-client/http.(*Service).Events
        /home/henridf/go/pkg/mod/github.com/attestantio/go-eth2-client@v0.7.1/http/events.go:62 +0x47b
```

--- a/services/blocks/standard/handler.go
+++ b/services/blocks/standard/handler.go
@@ -56,7 +56,6 @@ func (s *Service) OnBeaconChainHeadUpdated(
 	md, err := s.getMetadata(ctx)
 	if err != nil {
 		log.Error().Err(err).Msg("Failed to obtain metadata")
-		md.MissedSlots = append(md.MissedSlots, slot)
 		return
 	}
 

